pkg/obfuscate/sql_tokenizer.go

```diff
@@ -473,12 +477,14 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 			if kind == DollarQuotedFunc {
 				// this is considered an embedded query, we should try and
 				// obfuscate it
+				tok = tok[len(FuncTag):]
```

Comment:
I'm not sure I understand why you've changed the behaviour of `scanDollarQuotedString` to return the delimiters. It creates an inconsistency with `scanString` and other analogous functions. They all do _not_ return the delimiters. Furthermore, you remove them here just to add them back again.

Reply:
`scanString` does return the delimiter now.
For example, this test passes:

```
func TestScanString(t *testing.T) {
	assert := assert.New(t)
	query := `"foobar"`
	tok := NewSQLTokenizer(query, false, nil)
	tok.advance()
	tok.advance()
	_, bs := tok.scanString('"', DoubleQuotedString)
	assert.Equal(`"foobar"`, string(bs))
}
```

The reason is that we want to keep string delimiters in the obfuscated query. All of the functions should include the quotes now.

Unfortunately in the case of `DollarQuotedFunc`s, we need to strip the delimiters temporarily so we can obfuscate the embedded function before putting the delimiters back on.

Reply:
> The reason is that we want to keep string delimiters in the obfuscated query. All of the functions should include the quotes now.

I see. That makes sense.

> Unfortunately in the case of DollarQuotedFuncs, we need to strip the delimiters temporarily so we can obfuscate the embedded function before putting the delimiters back on.

Ouch. That makes sense. Maybe we can document it somewhere in code.

Thanks for explaining.